tests/test_debugging.py

```diff
@@ -437,7 +439,11 @@ def circuit(add_bad_snapshot: bool):
         assert result["execution_results"] == expected["execution_results"]
 
         # Make sure shots are overridden correctly
-        result = qml.snapshots(circuit)(add_bad_snapshot=False, shots=200)
+        with pytest.warns(
+            PennyLaneDeprecationWarning,
+            match="'shots' as an argument to the quantum function is deprecated",
+        ):
+            result = qml.snapshots(circuit)(add_bad_snapshot=False, shots=200)
```

Comment:
```suggestion
        result = qml.set_shots(shots=200)(qml.snapshots(circuit))(add_bad_snapshot=False)
```

Reply:
This should be good @astralcai 
Note that qml.snapshots returns the wrapped `get_snapshots` function so can't be easily applied by `set_shots`.
Honestly I feel the original testing line here already lost its meaning with our deprecation; being marked as deprecated and waiting for removal might be better

Reply:
wouldn't you apply `set_shots` on the circuit first, and then apply `snapshots` on the updated circuit?